content/chapters/compute/lecture/slides.md

```diff
@@ -0,0 +1,1161 @@
+20733 20661 147004 845832   05:43:40 Sl+                       \_ node
+20796 20733 119336 959952   05:43:38 Sl+                           \_ node
+21149 20733 41080 581504    04:31:20 Sl+                           \_ node
+20672     1   176    988    05:43:40 Ss   init
+20673 20672   176    988    05:43:40 S     \_ init
+20674 20673 41556 585172    05:43:40 Ssl+      \_ node
+20701     1   176    988    05:43:40 Ss   init
+20702 20701   176    988    05:43:40 S     \_ init
+20703 20702 48060 590308    05:43:40 Ssl+      \_ node
+20915     1   176    988    05:24:46 Ss   init
+20916 20915   176    988    05:24:46 S     \_ init
+20917 20916  5268  10188    05:24:46 Ss        \_ bash
+20964 20917 19840 516296    05:24:32 S+            \_ python3
+20981     1   176    988    05:22:54 Ss   init
+20982 20981   176    988    05:22:54 S     \_ init
+20983 20982  5100  10056    05:22:54 Ss+       \_ bash
+```
+
+----
+
+### Process Attributes - from Kernel Space
+
+* PCB (Process Control Block):
+    * Data structure in the kernel - `task_struct`
+    * All information about the process (from `ps`, `/proc`):
+
+----
+
+### Process Attributes - from Kernel Space
+
+The most important fields in `task_struct`:
+```c [2 | 4 | 6 | 8 - 9 | 11 - 13 | 15 - 16 | 18 - 19 | 21 | 23 | 25 - 26 | 28 - 29]
+struct task_struct {
+        unsigned int                    __state;
+
+        void                           *stack;
+
+        unsigned int                    flags;
+
+        int                             on_cpu;
+        int                             prio;
+
+        /* Scheduler information */
+        struct sched_entity             se;
+        const struct sched_class        *sched_class;
+
+        /* The VAS: memory mappings */
+        struct mm_struct                *mm;
+
+        int                             exit_state;
+        int                             exit_code;
+
+        pid_t                           pid;
+
+        struct task_struct __rcu        *parent;
+
+        /* Child processes */
+        struct list_head                children;
+
+        /* Open file information */
+        struct files_struct             *files;
+};
+```
+
+---
+
+### `fork`
+
+* Spawns child process
+* Child process starts execution right after the call to `fork`
+* Returns twice
+    * `0` in child process
+    * Child's PID in parent process
+[Quiz](./quiz/process-creation.md)
+* Initially child and parent share the same PAS (Physical Address Space)
+* Child's writable pages are marked **copy-on-write**
+
+----
+
+### Copy-on-write
+
+When `fork` returns, the child inherits the parent's page table.
+
+![Copy-on-write after fork()](./media/copy-on-write-initial.svg)
+
+----
+
+### Copy-on-write
+
+* Writable pages are **copied and modified** when written: `demo/copy-on-write/copy_on_write_overhead.c`.
+* [Quiz](./quiz/sections-always-shared.md)
+
+![Copy-on-write after Write](./media/copy-on-write-final.svg)
+
+----
+
+### Copy-on-write Overheard
```

Comment:
This doesn't render properly.

Reply:
It seems the `$` character was somehow interpreted either as HTML or as a genuine command such as `curl` whose output was embedded in the rendering :)))

Moral of the story: don't use backticks instead of `$(...)` around commands.